src/components/TodoList/TodoList.tsx

```diff
@@ -1 +1,37 @@
-export const TodoList = () => {};
+import { TodoInfo } from '../TodoInfo';
+
+import usersFromServer from '../../api/users';
+
+import {
+  Todo,
+  User,
+  FullTodoInfo,
+} from '../../react-app-env';
+
+type Props = {
+  todos: Todo[]
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  const prepareTodos = (users: User[], todosArr: Todo[]): FullTodoInfo[] => {
+    return todosArr.map(todo => ({
+      ...todo,
+      user: users.find(user => user.id === todo.userId),
+    }));
+  };
+
+  const preparedTodos = prepareTodos(
+    usersFromServer,
+    todos,
+  );
```

Comment:
You can do it once in parent component. It's not good idea to call api  twice every time you render the list (do it only in parent)